pygount/analysis.py

```diff
@@ -27,7 +27,11 @@
 from pygount.common import mapped_repr
 from pygount.git_storage import GitStorage, git_remote_url_and_revision_if_any
 
-GIT_REPO_REGEX = re.compile(r"^(https?://|git@)")
+HAS_URL_PREFIX = re.compile(r"^(https?://)")
+_ALLOWED_GIT_PLATFORMS = ["github.com", "bitbucket.org", "gitlab.com"]
+GIT_REPO_REGEX = re.compile(
+    r"^(https?://|git@)({})/[\w-]+/[\w-]+".format("|".join(map(re.escape, _ALLOWED_GIT_PLATFORMS)))
```

Comment:
Not sure of `\w-` is enough, there might be all kinds of cases where this could fail for legitimate URLs, e.g. when using URL escaping based on `%`. Maybe we should go with something more lenient, that would accept a few non-URLs. In that case, it will fail later anyway when trying to interact with the URL, which should cause some system function to fail and report a somewhat descriptive error message. Which would be good enough from my point of view.

Also, we should be able to avoid the `format()` using f-strings. How about this?
```python
_ALLOWED_GIT_PLATFORM_CHOICES_PATTERN = "|".join(map(re.escape, _ALLOWED_GIT_PLATFORMS)
GIT_REPO_REGEX = re.compile(
    fr"^(https?://|git@)({_ALLOWED_GIT_PLATFORM_CHOICES_PATTERN})/[^/]+/[^/]+"
)
```